src/components/WalletModal/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { ReactElement, useState } from 'react'
+import ReactModal from 'react-modal'
+import styled from 'styled-components'
+import { KARELIA } from '../../utils/css'
+import metamask from './icons/metamask.png'
+import keystone from './icons/keystone.png'
+import coinbase from './icons/coinbase.png'
+import walletconnect from './icons/walletconnect.png'
+
+const customStyles = {
+    overlay: {
+        backgroundColor: 'rgba(0, 0, 0, 0.3)',
+    },
+    content: {
+        border: 'none',
+        width: '528px',
+        height: '558px',
+        top: '50%',
+        left: '50%',
+        right: 'auto',
+        bottom: 'auto',
+        borderRadius: '20px',
+        transform: 'translate(-50%, -50%)',
+        fontFamily: `${KARELIA}`,
+    },
+}
+
+const StyledModalContent = styled.div`
+    .contentContainer {
+        display: flex;
+        flex-direction: column;
+        min-height: 100%;
+        height: 440px;
+        padding: 0px 20px;
+    }
+
+    .closeButton {
+        background-color: transparent;
+        border: none;
+    }
+
+    .modalHeader {
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+    }
+
+    .walletOptions {
+        flex: 1;
+        display: flex;
+        flex-direction: column;
+        overflow: auto;
+        min-height: 100%;
+    }
+
+    .walletOption {
+        background-color: #f1f4f7;
+        margin-top: 8px;
+        margin-bottom: 8px;
+        display: flex;
+        align-items: center;
+        flex: 1;
+        padding: 0px 32px;
+        border-radius: 8px;
+        justify-content: space-between;
+        font-size: 16px;
+        font-family: IBM Plex Sans;
+        font-weight: bold;
+    }
+
+    .walletOption:hover {
+        background-color: #dfe3e8;
+    }
+`
+
+const UnstyledWalletModal = ({ button }: { button?: ReactElement }) => {
+    const [modalIsOpen, setModalIsOpen] = useState(false)
+
+    const openModal = () => {
+        setModalIsOpen(true)
+    }
+
+    const closeModal = () => {
+        setModalIsOpen(false)
+    }
+
+    return (
+        <div>
+            {button ? (
+                React.cloneElement(button, { onClick: openModal })
+            ) : (
+                <button onClick={openModal} className="button">
+                    Connect a Wallet
+                </button>
+            )}
```

Comment:
We can actually go with just the button. No need to prepare it for different use cases. #kiss

Reply:
Gotcha! Thanks for the pointer!

Reply:
We can also apply this to #11… 🤔